### ISSUE TITLE
Smarter releasing and optional "parking"

### ISSUE THREAD

**Issue by WolfieWaffle** [AJCStriker/Movecraft-3#151]
This is a feature suggestion that I think would be useful. What I want is the ability to set a config option to use "parking" of ships and make it so there wont be un-piloted frozen floating ships everywhere. I don't really know how exactly to explain it, so here's just how it would work. I hope it's not too long and boring!

NO ABANDONED FLOATING SHIPS:
Ships could become "fake-piloted." This wouldn't be driving by themselves or coming to life magically, but being piloted by a fake player or similar so that they would still sink. Basically, Imagine this: A player jumps out of their ship. After the timeout, the ship is piloted and if it's an airship, it is just floating there. Since it is no longer considered a ship, it will not sink even if it takes damage, and the blocks will just float there forever, clogging up the server. This has been an issue for me, and probably others as well.

So instead of just being un-piloted when you release them, they would become "fake piloted." This would mean that they would still sink if damaged for a while, so they wouldn't just float around and junk up the world (this was a problem I had). You could configure the following:

- Timeout for being "auto piloted" aka how long it would still be sinkable after a player has stopped piloting it. Once it has timed out, you could configure it to:
    - Sink. It would float for a while, damageable or re-pilot able, but then fall downwards (there no player to keep it in the air, after all!)
    - Stay in the air and stop being a ship. This is the behavior at the moment, where the blocks are no longer part of a ship and just float there.

This way, ships would not be floating around randomly. One problem I had was that I would somehow unpilot the ship and try to re pilot it, but it was too damaged and the ship would neither sink or be piloted. With this feature, the ship would be "fake piloted" for a while, and realize that is should sink, rather that just float in the air, frozen, which happens to me a lot.

LANDING/PARKING
Of course, with this option, ships would unnecessary be fake-piloted even when they are landed on the ground or on a dock, which could cause lag. So a sort of "parking" system would be implemented. For airships, this is simple. If they are released and there are solid blocks below them, they will not be fake-piloted - they will just follow the normal behavior and turn into regular non-flying blocks.

For regular ships, this is more complicated. If they are abandoned in the ocean, it would be bad to have them fake-piloted forever, since it would cause lag and ships would build up. But making them sink doesn't make sense, and making them released (non-ship) would still create abandoned ships everywhere. I think the best option for this would to be sinking (and releasing when they hit the ground, no fake-piloting) because IRL a ship would probably eventually sink due to wind and disrepair.

To land/park a water ship, you would probably have it within range of a certain amount of solid blocks, such as the shore or a dock. In the ocean, with no solid blocks around it, it would not be released immediately, sinking (or another option from the first part of this section) after the timeout for the fake-piloting.

The parking/landing would not have a command or anything, it's just automatic. If it is not landed or docked, it will continue to be sinkable for a while, else it will be released as normal.

WHAT THIS WOULD DO:
It would be an optional setting to allow ship to sink if abandoned in an unsafe place. If you jump off an airship, it will sink. If for some reason you release your ship and immediately after somebody blows up half of it, it will sink rather than becoming frozen. If you don't properly dock your ship, there will be consequences.

I realize it might make the server slow. So it would be optional, and have a configurable fake-pilot timeout. It would release the ships eventually, so there would be no fake sip buildup. Hopefully, it would prevent the world from being clogged up by random piloted ships and make it the plugin more enjoyable.

TL;DR
Ships will no longer float around uselessly when released. You must dock or land them.

ENDNOTES:
I'm not trying to change this plugin or anything, just trying to solve a problem. Everything should be optional and configurable, like always. I know that this might seem like a giant wall of text, but the idea is quite simple and I was just trying to provide detail. Again, it's just an option fix to a problem that I'm sure other people have had.

Above all, keep making this amazing plugin and don't get mad at me! I'm just trying to help :)
